src/main/java/com/iota/iri/service/API.java

```diff
@@ -100,6 +100,8 @@
     private HashMap<InetAddress,AtomicInteger> broadcastStoreCounters;
     private AtomicLong broadcastStoreTimer;
 
+    private ConcurrentHashMap<Hash, Boolean> previousEpochsSpentAddresses;
```

Comment:
As we don't modify the map after creation, how about we make this a regular immutable Map?